x-pack/plugins/observability_ai_assistant/server/functions/esql/correct_common_esql_mistakes.test.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import dedent from 'dedent';
+import { noop } from 'lodash';
+import { correctCommonEsqlMistakes } from './correct_common_esql_mistakes';
+
+describe('correctCommonEsqlMistakes', () => {
+  const fakeLogger = {
+    debug: noop,
+  } as any;
+
+  function renderQuery(query: string) {
+    return '```esql\n' + dedent(query) + '\n```';
+  }
+
+  function expectQuery(input: string, expectedOutput: string) {
+    expect(correctCommonEsqlMistakes(renderQuery(input), fakeLogger)).toEqual(
+      renderQuery(expectedOutput)
+    );
+  }
+
+  it('replaces aliasing via the AS keyword with the = operator', () => {
+    expectQuery(`FROM logs-* | STATS COUNT() AS count`, 'FROM logs-*\n| STATS count = COUNT()');
+    expectQuery(
+      `FROM logs-* | STATS AVG(transaction.duration.histogram) AS avg_request_latency, PERCENTILE(transaction.duration.histogram, 95) AS p95`,
+      `FROM logs-*
+      | STATS avg_request_latency = AVG(transaction.duration.histogram), p95 = PERCENTILE(transaction.duration.histogram, 95)`
+    );
+
+    expectQuery(
+      `FROM traces-apm*
+      | WHERE @timestamp >= NOW() - 24 hours
+      | STATS AVG(transaction.duration.us) AS avg_duration, SUM(success) / (SUM(success) + SUM(failure)) AS success_rate, COUNT(*) AS total_requests BY service.name`,
+      `FROM traces-apm*
+      | WHERE @timestamp >= NOW() - 24 hours
+      | STATS avg_duration = AVG(transaction.duration.us), success_rate = SUM(success) / (SUM(success) + SUM(failure)), total_requests = COUNT(*) BY service.name`
+    );
+  });
+
+  it('replaces " or \' escaping in FROM statements with backticks', () => {
+    expectQuery(`FROM "logs-*" | LIMIT 10`, 'FROM `logs-*`\n| LIMIT 10');
```

Comment:
I spent more time than I'd like to admit trying to figure out how JavaScript would be able to tell you which "tick style" was used to create the string before I realized you're using ` to escape the " ...